client/app/src/main/java/com/example/mrides/Notification/MatchingMessagingService.java

```diff
@@ -60,21 +66,31 @@ public void onMessageReceived(RemoteMessage remoteMessage) {
         notificationBuilder.setContentIntent(pendingIntent);
         NotificationManager notificationManager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
         notificationManager.notify(0,notificationBuilder.build());
+        Notify(remoteMessage.getData());
     }
 
 
+
     @Override
     public void attach(ActivityObserver observerToAdd) {
 
+        if(observers.contains(observerToAdd)){
```

Comment:
There should be a space between the closing parenthesis and opening curly bracket.